subsystem/kubeapi_test.go

```diff
@@ -0,0 +1,219 @@
+	Expect(cluster.Name).Should(Equal(spec.Name))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.BaseDNSDomain).Should(Equal(spec.BaseDNSDomain))
+	Expect(cluster.ClusterNetworkCidr).Should(Equal(spec.ClusterNetworkCidr))
+	Expect(cluster.ClusterNetworkHostPrefix).Should(Equal(spec.ClusterNetworkHostPrefix))
+	Expect(cluster.ServiceNetworkCidr).Should(Equal(spec.ServiceNetworkCidr))
+	Expect(cluster.IngressVip).Should(Equal(spec.IngressVip))
+	Expect(cluster.SSHPublicKey).Should(Equal(spec.SSHPublicKey))
+	Expect(swag.BoolValue(cluster.VipDhcpAllocation)).Should(Equal(spec.VIPDhcpAllocation))
+	Expect(cluster.HTTPProxy).Should(Equal(spec.HTTPProxy))
+	Expect(cluster.HTTPSProxy).Should(Equal(spec.HTTPSProxy))
+	Expect(cluster.NoProxy).Should(Equal(spec.NoProxy))
+	Expect(swag.BoolValue(cluster.UserManagedNetworking)).Should(Equal(spec.UserManagedNetworking))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.AdditionalNtpSource).Should(Equal(spec.AdditionalNtpSource))
+	Expect(cluster.PullSecret).Should(Equal(pullSecret))
+}
+
+func cleanUP(ctx context.Context, client k8sclient.Client) {
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Image{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
+}
+
+var _ = Describe("cluster installation", func() {
+	if !Options.EnableKubeAPI {
+		return
+	}
+
+	ctx := context.Background()
+
+	waitForClusterReconcileTimeout := 10
+
+	secretRef := &corev1.SecretReference{}
+
+	generateFAPostStepReply := func(h *models.Host, freeAddresses models.FreeNetworksAddresses) {
+		fa, err := json.Marshal(&freeAddresses)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = agentBMClient.Installer.PostStepReply(ctx, &installer.PostStepReplyParams{
+			ClusterID: h.ClusterID,
+			HostID:    *h.ID,
+			Reply: &models.StepReply{
+				ExitCode: 0,
+				Output:   string(fa),
+				StepID:   string(models.StepTypeFreeNetworkAddresses),
+				StepType: models.StepTypeFreeNetworkAddresses,
+			},
+		})
+		Expect(err).To(BeNil())
+	}
+
+	secretRef = &corev1.SecretReference{
+		Namespace: Options.Namespace,
+		Name:      "pull-secret",
+	}
+	err := kubeClient.Get(ctx, types.NamespacedName{Namespace: Options.Namespace, Name: "pull-secret"}, &corev1.Secret{})
+	if apierrors.IsNotFound(err) {
+		deployPullSecretResource(ctx, kubeClient, secretRef.Name, pullSecret)
+	} else {
+		Expect(err).To(BeNil())
+	}
+
+	AfterEach(func() {
+		cleanUP(ctx, kubeClient)
+		clearDB()
+	})
+
+	It("[kube-api]deploy cluster with hosts and wait for preparing-for-installation", func() {
+		spec := getDefaultClusterSpec(secretRef)
+		deployClusterCRD(ctx, kubeClient, spec)
+		key := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      spec.Name,
+		}
+		cluster := getClusterFromDB(ctx, kubeClient, db, key, waitForClusterReconcileTimeout)
```

Comment:
again until update ticket is resolved (maybe we have other update issues as well?)  i'm suggesting do if after the cluster starting the instalaltion.

Reply:
why? I am not waiting for any state yet, here I just validating the creation of the cluster with the correct parameters

Reply:
I mean that if this test validates cluster installation then comparing the values is not relevant for this test. we are going to move to a new cluster definition [MGMT-3721](https://issues.redhat.com/browse/MGMT-3721) so i'm suggesting not to make tests that will break on translation, at least not for now.